authors/apps/notifications/tests/test_notifications.py

```diff
@@ -0,0 +1,168 @@
+"""Module houses tests for notifications"""
+from rest_framework.test import APIClient
+from rest_framework import status
+from ..models import Notification
+from ...authentication.tests.test_setup import BaseSetUp
+from django.test import TestCase
+
+
+class SetUp(TestCase):
+    """Base setup class to be shared by other test classess"""
+
+    def setUp(self):
+        """Initialize test data"""
+        self.base = BaseSetUp()
+        self.login_data = {
+            'username': self.base.reg_data['username'],
+            'password': self.base.reg_data['password']
+        }
+        # login a test user and generate token
+        self.login_res = self.base.client.post(
+            'api/users/login',
+            self.login_data,
+            format="json")
+        self.token = self.login_res.data['Token']
+        # pass token to test client to authorize api access
+        self.base.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
```

Comment:
Please pass this token at every endpoint as you create a response. Having it here does not assign it to every response.

Reply:
I had done that then I decided to refactor and add a setup class where all the other test classes inherit from. Doesn't that mean all the tests can access the token? Also from the drf documentation, it's advisable to add it to your setup.